CMakeLists.txt

```
@@ -158,12 +158,11 @@ message(STATUS
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
 if(JPEGXL_STATIC)
-  set(CMAKE_FIND_LIBRARY_SUFFIXES .a)
   set(BUILD_SHARED_LIBS 0)
   # Clang developers say that in case to use "static" we have to build stdlib
   # ourselves; for real use case we don't care about stdlib, as it is "granted",
   # so just linking all other libraries is fine.
-  if (NOT APPLE)
+  if (UNIX AND NOT APPLE)
```

Are there any other non-UNIX targets besides MSVC that might be influenced by this change?
It might perhaps be better to do
```suggestion
  if (NOT MSVC AND NOT APPLE)
```
?

---

If you're on Windows and compiling in a Cygwin or MSYS the UNIX variable is set to `TRUE` so I believe this is what you would want.

In CMake one of the places that does `set(UNIX 1)` is in https://gitlab.kitware.com/cmake/cmake/-/blob/master/Modules/Platform/UnixPaths.cmake#L15 which is included in the CYWIN module at https://gitlab.kitware.com/cmake/cmake/-/blob/master/Modules/Platform/CYGWIN.cmake#L64 and MSYS includes that at https://gitlab.kitware.com/cmake/cmake/-/blob/master/Modules/Platform/MSYS.cmake#L2 .

The docs for CMake aren't explicit on what platforms count as UNIX but this change would prevent MSVC and Apple builds from adding these flags and would keep the same behavior for all other platforms.

---

Since our time zones don't have much of any overlap for working hours I want to be clear that I'm fine with the proposed change being committed. I just wanted to make sure I gave enough information for you all to make an informed choice on what the values should be.

Thanks for the review!

---

The Cygwin-based MSYS subsystem sets it but MinGW (“Windows-GNU”) does not: https://gitlab.kitware.com/cmake/cmake/-/blob/master/Modules/Platform/Windows-GNU.cmake

So I think `NOT MSVC` would indeed be preferable here.